src/zmalloc.c

```diff
@@ -87,10 +87,22 @@ void zlibc_free(void *ptr) {
 #define dallocx(ptr,flags) je_dallocx(ptr,flags)
 #endif
 
-#define update_zmalloc_stat_alloc(__n) atomicIncr(used_memory,(__n))
-#define update_zmalloc_stat_free(__n) atomicDecr(used_memory,(__n))
+#define update_zmalloc_stat_alloc(__n) used_memory_tls[thread_index] += (__n)
+#define update_zmalloc_stat_free(__n) used_memory_tls[thread_index] -= (__n)
 
-static serverAtomic size_t used_memory = 0;
+/* A thread-local storage which keep the current thread's index in the
+ * used_memory_tls array. */
+static __thread int thread_index;
+/* MAX_THREADS_NUM = IO_THREADS_MAX_NUM(128) + BIO threads(3) + main thread(1). */
+#define MAX_THREADS_NUM 132
+static size_t used_memory_tls[MAX_THREADS_NUM];
```

Comment:
```suggestion
static unsigned long long used_memory_tls[MAX_THREADS_NUM];
```
The BIO threads will underflow, since it's only releasing memory, and the main thread will likely overflow if there is a BIO thread freeing memory. Although I don't think this is technically an issue, I think it would be clearer to indicate that with a signed variable then leaving it unsigned.

Reply:
Thanks for the suggestion, changed.